be/src/exprs/array_functions.tpp

```diff
@@ -1203,4 +1204,75 @@ private:
     }
 };
 
+// Todo:support datatime/data
+template <LogicalType Type>
+class ArrayGenerate {
+public:
+    using InputColumnType = RunTimeColumnType<Type>;
+    using InputCppType = RunTimeCppType<Type>;
+    static StatusOr<ColumnPtr> process(FunctionContext* ctx, const Columns& columns) {
+        RETURN_IF_COLUMNS_ONLY_NULL(columns);
+        DCHECK(columns.size() == 3);
+
+        auto num_rows = columns[0]->size();
+
+        // compute nulls first. if any input is null, then output is null
+        NullColumnPtr nulls;
+        for (auto& column : columns) {
+            if (column->has_null()) {
+                auto nullable_column = down_cast<NullableColumn*>(column.get());
+                if (nulls == nullptr) {
+                    nulls = std::static_pointer_cast<NullColumn>(nullable_column->null_column()->clone_shared());
+                } else {
+                    ColumnHelper::or_two_filters(num_rows, nulls->get_data().data(),
+                                                 nullable_column->null_column()->get_data().data());
+                }
+            }
+        }
+
+        auto array_offsets = UInt32Column::create(0);
+        auto array_elements = ColumnHelper::create_column(TypeDescriptor(Type), true, false, 0);
+
+        auto offsets = array_offsets.get();
+        auto elements = down_cast<NullableColumn*>(array_elements.get());
+
+        TRY_CATCH_BAD_ALLOC(offsets->reserve(num_rows + 1));
```

Comment:
offsets is fixed length column whose length does not exceeds 4096,  no need to use TRY_CACHE_BAD_ALLOC wrap it.

Reply:
I will fix it